osquery/tables/system/windows/registry.cpp

```diff
@@ -72,6 +72,21 @@ void explodeRegistryPath(const std::string& path,
   rKey = osquery::join(toks, kRegSep);
 }
 
+Status sanitizeRegistryStrings(const DWORD dataType,
+                               BYTE* dataBuff,
+                               DWORD dataSize) {
+  if (!dataBuff ^ (dataSize == 0)) {
+    return Status(1, "Invalid registry data to sanitize");
+  }
+  if (dataBuff && dataSize != 0) {
+    if (dataSize > 0 &&
```

Comment:
Additionally, would it be possible to bring this logical check up into the first `if`? Seems like this could all be done with 1 `if` check?